### ISSUE TITLE
The `@liveblocks/node` client crashes when secret key is invalid

### ISSUE THREAD

**Issue by nvie** [liveblocks/liveblocks#1381]
**Describe the bug**

When the passed-in secret key is invalid, this isn't reported with a useful runtime error. Instead, it fails because of a `"undefined is not an object (evaluating data.data.map")`.

**To Reproduce**

Example snippet:

```ts
const client = new Liveblocks({ secret: "not valid" });
await client.getRooms();  // crash
```

**Expected behavior**

A clear runtime error, informing the user about the incorrectly configured secret key.

**Comment by nimeshnayaju** [comment]
This would indeed be useful. Failing with `"undefined is not an object (evaluating data.data.map")` is far from ideal. We seem to miss checking the status of `res` before using parsing it as JSON. We do this for other methods but missed for `getRooms` method.


```ts
const res = await this.get(path, queryParams);

const data = (await res.json()) as {
  nextPage: string | null;
  data: RoomInfoPlain[];
};
```

**Comment by saiprakash2** [comment]
Hi Guys,
The `getRooms` method is missing this part of the code. This error is handled in other methods but it is missing in `getRooms` method.
If you are good with it I can raise the PR 🙂

`    if (!res.ok) {
      const text = await res.text();
      throw new LiveblocksError(res.status, text);
    }`

**Comment by nimeshnayaju** [comment]
Hi @saiprakash2 , thanks for offering to help and your willingness to contribute to Liveblocks.  I wanted to let you know that we recently identified the issue as well and submitted a PR to address it. You can view the PR [here](https://github.com/liveblocks/liveblocks/pull/1383).

We really appreciate your initiative to offer help. If you have any other suggestions or find more areas where you think improvements can be made, please feel free to raise an issue or a PR. We are always looking for ways to improve!

**Comment by nvie** [comment]
Closing because this is fixed in [v1.9.3](https://github.com/liveblocks/liveblocks/releases/tag/v1.9.3).
